### ISSUE TITLE
ScrollableTextLayout does not respect content_valign

### ISSUE THREAD

**Issue by Shildifreak** [pyglet/pyglet#360]
I was trying to create a scrollable chat window, which fills from the bottom, but if there are more messages than can be displayed gets cut off at the top. So as long as the user doesn't scroll up the end of the document is always aligned with the bottom of the viewing area.
If I understand the documentation* correctly, this should be possible using a ScrollableTextLayout and setting:
```python
l = pyglet.text.layout.ScrollableTextLayout(...)
l.content_valign = "bottom"
l.view_y = l.height - l.content_height
```
However, in practice when the content is smaller than the viewing area it gets aligned to the top, despite content_valign telling it to do otherwise.

All my tests were done on the latest current release 1.5.15.

*documentation on `ScrollableTextLayout.view_y`:
```python
"""
Vertical scroll offset.

    The initial value is 0, and the top of the text will touch the top of the
    layout bounds **(unless the content height is less than the layout height,
    in which case `content_valign` is used)**.

    A negative value causes the text to "scroll" upwards.  Values outside of
    the range ``[height - content_height, 0]`` are automatically clipped in
    range.

    :type: int
"""
```

If someone else finds this and needs a quick workaround, I am for now bypassing the v_view range check as follows:
```python
# workaround
l.content_valign = "top"
l.top_group.view_y = l.height - l.content_height
```

**Comment by benmoran56** [comment]
Thanks for opening the issue. I have not used this extensively myself, but the `content_valign` setting _does_ work correctly when using the `IncrementalTextLayout`. I'm not sure yet if this is a normal or not. In any case, there is either a failure of documentation, or some bug in the code.

**Comment by benmoran56** [comment]
On further testing, the standard `TextLayout` class also respects the `content_valign`, so it looks like just a bug in the `ScrollableTextLayout`.

**Comment by benmoran56** [comment]
I'm working on a fix for this at the moment. I'm actually rewriting the ScrollableTextLayout classes a bit to clean things up. This is taking a little more time, but I hope to have it done soon.

**Comment by Shildifreak** [comment]
Great, I'm looking forward to it. (But don't worry, if something more important comes up, it's not a pressing issue on my side.)

**Comment by benmoran56** [comment]
This is now fixed in the latest 1.5.16 release. Please give that a try, and feel free to follow up here if you're still seeing issues.
